solidity/random-beacon/contracts/libraries/DKG.sol

```diff
@@ -303,75 +295,86 @@ library DKG {
         require(
             block.number >=
                 (T_init +
-                    (submitterMemberIndex - 1) *
+                    (result.submitterMemberIndex - 1) *
                     self.parameters.resultSubmissionEligibilityDelay),
             "Submitter not eligible"
         );
 
-        require(groupPubKey.length == 128, "Malformed group public key");
+        require(result.groupPubKey.length == 128, "Malformed group public key");
 
         require(
-            misbehavedMembersIndices.length <= groupSize - signatureThreshold,
+            result.misbehavedMembersIndices.length <=
+                groupSize - signatureThreshold,
             "Unexpected misbehaved members count"
         );
 
-        if (misbehavedMembersIndices.length > 1) {
-            for (uint256 i = 1; i < misbehavedMembersIndices.length; i++) {
+        if (result.misbehavedMembersIndices.length > 1) {
+            for (
+                uint256 i = 1;
+                i < result.misbehavedMembersIndices.length;
+                i++
+            ) {
                 require(
-                    misbehavedMembersIndices[i - 1] <
-                        misbehavedMembersIndices[i],
+                    result.misbehavedMembersIndices[i - 1] <
+                        result.misbehavedMembersIndices[i],
                     "Corrupted misbehaved members indices"
                 );
             }
         }
 
-        uint256 signaturesCount = signatures.length / 65;
-        require(signatures.length >= 65, "Too short signatures array");
-        require(signatures.length % 65 == 0, "Malformed signatures array");
+        uint256 signaturesCount = result.signatures.length / 65;
+        require(result.signatures.length >= 65, "Too short signatures array");
+        require(
+            result.signatures.length % 65 == 0,
+            "Malformed signatures array"
+        );
         require(
-            signaturesCount == signingMembersIndices.length,
+            signaturesCount == result.signingMembersIndices.length,
             "Unexpected signatures count"
         );
         require(signaturesCount >= signatureThreshold, "Too few signatures");
         require(signaturesCount <= groupSize, "Too many signatures");
 
         bytes32 resultHash = keccak256(
             abi.encodePacked(
-                groupPubKey,
-                misbehavedMembersIndices,
+                result.groupPubKey,
+                result.misbehavedMembersIndices,
                 self.startBlock
             )
         );
 
         bytes memory current; // Current signature to be checked.
-
         bool[] memory usedMemberIndices = new bool[](groupSize);
+        signingMembers = new uint32[](signaturesCount);
+        address[] memory membersAddresses = sortitionPool.getIDOperators(
+            result.members
+        );
```

Comment:
We do not need to fetch the addresses of all members, addresses of signing group members should be enough. Do we know what is the gas cost difference when working with a real sortition pool? Fetching just signing member addresses will require at least one iteration through `signingMembersIndices` and building an array with their identifiers but maybe it's still cheaper than doing more SREADs on the sortition pool side.

Reply:
We don't have a real sortition pool plugged-in yet to make a fair comparison. Maybe we can leave a TODO and check this while doing the optimization work?

Reply:
Fine for me, let's leave a TODO.

Reply:
See https://github.com/keep-network/keep-core/pull/2695/commits/52a1623ca0747be3944ad5e3fe060e3d9b6b05cf